wrangler-service/src/main/java/co/cask/wrangler/service/FileTypeDetector.java

```diff
@@ -46,7 +46,7 @@ public FileTypeDetector() {
       Scanner scanner = new Scanner(file);
```

Comment:
This is bad. It uses the default charset to decode the file. Please use the one with charset `public Scanner(File source, String charsetName)`

Reply:
Also, can't comment on line 44, but shouldn't do `new File(Explorer.class.getClassLoader().getResource("file.extensions").getFile())`, as that assume the `file.extensions` resource is available in file system, which is not always the case (it is ClassLoader implementation specific). Should use the `Scanner(InputStream source, String charsetName)` constructor instead, by opening the stream from the `URL` returned by the `getResource` method.

```java
try (InputStream is = getClass().getClassLoader().getResource("file.extensions").openStream()) {
  Scanner scanner = new Scanner(is, "UTF-8");
  ...
}
```